Sage/carbon
# fix(menu-item): maintain focus order when submenu child item is clicked

fix https://github.com/Sage/carbon/issues/5850

### Proposed behaviour


Ensure that when a `Submenu` child item is clicked any subsequent keyboard navigation focuses the
correct next or previous item.

![fixed-click-menu-item](https://user-images.githubusercontent.com/44157880/220167861-8004c71c-4e6b-4e2b-8df6-6c7e66d459ba.gif)


Ensures that the first `MenuItem` is only focused when `Submenu` is opened
via key press

![image](https://user-images.githubusercontent.com/44157880/220166965-0fa87d66-0849-4045-9be2-13b982d6ae5b.png)


### Current behaviour


Clicking a menu item and then keyboard navigating does not focus expected items

![boirked-click-menu-item](https://user-images.githubusercontent.com/44157880/220167962-a4162357-9145-4801-92ec-9cb81bcea979.gif)


First child item of submenu is focused when opening via mouseover

![image](https://user-images.githubusercontent.com/44157880/220167053-b13f2ad1-ab99-4ff8-810a-efa52d3f3b52.png)

### Checklist




#### QA


### Additional context



### Testing instructions


Open submenu (in iframe), click an item, use keyboard to navigate and confirm everything is great! 

Mouseover submenu item and confirm the first child is not auto focused, but that when you open via keypress it is

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
